src/web/src/asset_actions_POST.ecpp

```diff
@@ -100,59 +107,64 @@ bool database_ready;
-            if (!i.getMember("command", cmd)) {
-                throw std::runtime_error("expected command key in object");
+            if (!i.getMember ("command", cmd)) {
+                throw std::runtime_error ("expected command key in object");
             }
-            i.getMember("parameter", param);
-            commands.emplace_back(cmd, param);
+            i.getMember ("parameter", param);
+            commands.emplace_back (cmd, param);
         }
     }
     catch (const std::exception& e) {
-        std::string err = std::string("Error while parsing document: ") + e.what() + ".";
+        std::string err = std::string ("Error while parsing document: ") + e.what () + ".";
         log_error (err.c_str ());
+        log_info_audit ("Request POST asset_actions asset %s FAILED", item.c_str ());
         http_die ("bad-request-document", TRANSLATE_ME (err.c_str ()).c_str ());
     }
 
     // send request
-    zmsg_t *rqst = zmsg_new();
-    zmsg_addstr(rqst, "DO_COMMANDS");
-    zmsg_addstr(rqst, "");
-    zmsg_addstr(rqst, checked_id.c_str());
+    zmsg_t *rqst = zmsg_new ();
+    zmsg_addstr (rqst, "DO_COMMANDS");
+    zmsg_addstr (rqst, "");
+    zmsg_addstr (rqst, checked_id.c_str ());
     for (const auto &i : commands) {
-        zmsg_addstr(rqst, i.first.c_str());
-        zmsg_addstr(rqst, i.second.c_str());
+        zmsg_addstr (rqst, i.first.c_str ());
+        zmsg_addstr (rqst, i.second.c_str ());
     }
 
-    if (mlm_client_sendto(client, "fty-nut-command", "power-actions", nullptr, 5000, &rqst) != 0) {
+    if (mlm_client_sendto (client, "fty-nut-command", "power-actions", nullptr, 5000, &rqst) != 0) {
         std::string err = "Failed to send request to fty-nut-command.";
         log_error (err.c_str ());
+        log_info_audit ("Request POST asset_actions asset %s FAILED", item.c_str ());
         http_die ("internal-error", TRANSLATE_ME (err.c_str ()).c_str ());
     }
 
     // wait for reply
-    ZpollerGuard poller(zpoller_new(mlm_client_msgpipe(client), nullptr));
-    if (!zpoller_wait(poller, 5000)) {
-        std::string err = "zmq_poll() failed.";
+    ZpollerGuard poller (zpoller_new (mlm_client_msgpipe (client), nullptr));
+    if (!zpoller_wait (poller, 5000)) {
+        std::string err = "zmq_poll () failed.";
         log_error (err.c_str ());
+        log_info_audit ("Request POST asset_actions asset %s FAILED", item.c_str ());
         http_die ("internal-error", TRANSLATE_ME (err.c_str ()).c_str ());
     }
 
-    ZmsgGuard msg(mlm_client_recv(client));
-    ZstrGuard status(zmsg_popstr(msg));
-    ZstrGuard uuid(zmsg_popstr(msg));
-    if (!streq(status, "OK")) {
-        ZstrGuard errMsg(zmsg_popstr(msg));
-        std::string err = std::string("Request to fty-nut-command failed: ") + errMsg.get() + ".";
+    ZmsgGuard msg (mlm_client_recv (client));
+    ZstrGuard status (zmsg_popstr (msg));
+    ZstrGuard uuid (zmsg_popstr (msg));
+    if (!streq (status, "OK")) {
```

Comment:
Not sure that streq() support NULL arg. + ZstrGuard() usage. Be carefull here.

Reply:
I don't know as well, but that's what's been used in the system, this is no new code.. As per first comment, ignore whitespace changes to see added code. But for sure we can discuss fixing things in rest here, but it's not aim of this PR

Reply:
Note that effectively Jeff's question goes for implementation of `strcmp()` because in ZMQ stack we `#define streq(s1,s2)    (!strcmp ((s1), (s2)))`. The standard does not define `strcmp` behavior for `NULL` inputs, so anything is allowed to happen, from a sanity check to a segmentation fault, any answer is correct.

Reply:
Looking at czmq code at https://github.com/42ity/czmq/blob/v3.0.2-FTY-master/src/zmsg.c#L407 and https://github.com/42ity/czmq/blob/v3.0.2-FTY-master/src/zframe.c#L238 there are several scenarios that `zmsg_popstr()` would return `NULL`, and we do not check nor at least assert for such cases anywhere, so Jeff's concern is valid :)

Reply:
All that said, if stars align and reading (dereferencing) data from NULL pointer reads some garbage (or the OS-mapped zero-filled page for example), so the program does not crash on this read, the `strcmp()` would quickly and highly-probably find there is no `OK` in the `status` so decide to ... (whatever it needs the clause for).

Reply:
While the question is right, this is a wrong place for such a question. I think this should be promoted to a task to check RestAPI for such uses of possibly invalid recvd data

Reply:
Sounds correct.